src/vm/dllimport.cpp

```diff
@@ -3975,9 +3985,16 @@ static void CreateNDirectStubWorker(StubState*         pss,
 #else // UNIX_X86_ABI
         fMarshalReturnValueFirst = HasRetBuffArg(&msig);
 #endif // UNIX_X86_ABI
-
+#elif defined(_WIN32) && defined(_TARGET_AMD64_)
+        fMarshalReturnValueFirst = SF_IsCOMStub(dwStubFlags) && msig.GetReturnType() == ELEMENT_TYPE_VALUETYPE;
```

Comment:
`SF_IsCOMStub(dwStubFlags)` is called above and provided to the base class. Can we tie them together?